### DISCUSSION TITLE
Cross DAG Dependencies Information using Stable Rest API

### DISCUSSION THREAD

**Discussion by Stuti98** [apache/airflow#25460] [Q&A]
Is there a way to get information on all cross DAG dependencies in our Airflow instance using the Stable REST API? I see we can make use of the [https://airflow.apache.org/api/v1/dags/{dag_id}/tasks](https://airflow.apache.org/api/v1/dags/%7Bdag_id%7D/tasks) to check if there is a trigger task or wait task (ExternalTaskSensor) but how do we get the information as to which DAG it is triggering or waiting on?

**Comment by potiuk** [comment]
There is no such information in the API (currently). 

We are working on https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-48+Data+Dependency+Management+and+Data+Driven+Scheduling that (first version of) is going to be released in 2.4 (in a month or so). That should have some variant of what you want - namely being able to track dataset dependencies (for example see that one DAG/task produce a dataset while the other consumes it), but I don't think there is an information about what are cross-dag dependencies if they will not be done in "dataset" way..

Regarding plain Triggering the DAGs and dependencies there, Airlfow UI (as of 2.1) is able to extract the data from SerializedModel table in database (and visualise it), so theoretically there is a way to do it from a database. But there are no API REST endpoints to read or query it AFAIK. 

That woudl be a nice contribution though - if you would like to contribute PRs implementing such API endpoints, that woudl be a great contribution, so feel free.
